cirq-core/cirq/sim/density_matrix_simulator.py

```diff
@@ -294,6 +251,12 @@ class DensityMatrixStepResult(
             results, ordered by the qubits that the measurement operates on.
     """
 
+    @deprecated_parameter(
+        deadline='v0.16',
+        fix='Remove parameter `simulator` as it is no longer used.',
+        parameter_desc='simulator',
+        match=lambda args, kwargs: 'simulator' in kwargs or len(args) > 2,
+    )
```

Comment:
Does this need a deprecation ?

Reply:
It's a public class, so in principle we shouldn't change the constructor without a deprecation cycle. In reality there's no reason for anyone ever to create one of these outside the simulator though, so it's not likely anyone would be broken if we just deleted it.

Reply:
Did you have a preference? I can either delete this parameter now or leave it for next release. If the latter then this is ready to merge.

Reply:
@MichaelBroughton is there anything holding this up now? I have a couple more deprecation PRs that depend on this one. I'd like this to go in soon if possible.